sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/implementation/ChangeFeedProcessorBuilderImpl.java

```diff
@@ -209,6 +214,92 @@ public boolean isStarted() {
             });
     }
 
+    /**
+     * Returns a list of states each representing one scoped worker item.
+     * <p>
+     * An empty list will be returned if the processor was not started or no lease items matching the current
+     *   {@link ChangeFeedProcessor} instance's lease prefix could be found.
+     *
+     * @return a list of states each representing one scoped worker item.
+     */
+    @Override
+    public Mono<List<ChangeFeedProcessorState>> getCurrentState() {
+        List<ChangeFeedProcessorState> earlyResult = new ArrayList<>();
+
+        if (this.leaseStoreManager == null || this.feedContextClient == null) {
+            return Mono.just(earlyResult);
+        }
+
+        return this.leaseStoreManager.getAllLeases()
+            .flatMap(lease -> {
+                ChangeFeedOptions options = new ChangeFeedOptions()
+                    .setMaxItemCount(1)
+                    .setPartitionKeyRangeId(lease.getLeaseToken())
+                    .setStartFromBeginning(true)
+                    .setRequestContinuation(lease.getContinuationToken());
+
+                return this.feedContextClient.createDocumentChangeFeedQuery(this.feedContextClient.getContainerClient(), options)
+                    .take(1)
+                    .map(feedResponse -> {
+                        final String pkRangeIdSeparator = ":";
+                        final String segmentSeparator = "#";
+                        final String lsnPropertyName = "_lsn";
+                        final String tsPropertyName = "_ts";
+
+                        String sessionTokenLsn = feedResponse.getSessionToken();
```

Comment:
NIT: final

Reply:
Same for several of the local variables below

Reply:
Though these have one assignment, it will look a bit odd when compared with similar patterns in the rest of the implementation (not just CFP).
And since they are not consumed by any Reactor code as arguments down the execution path which requires as such, they don't really need to be final...